src/steps/generic.rs

```diff
@@ -369,15 +369,33 @@ pub fn run_gcloud_components_update(ctx: &ExecutionContext) -> Result<()> {
     let gcloud = require("gcloud")?;
 
     if gcloud.starts_with("/snap") {
-        Ok(())
-    } else {
-        print_separator("gcloud");
+        return Ok(());
+    }
 
-        ctx.run_type()
-            .execute(gcloud)
-            .args(["components", "update", "--quiet"])
-            .status_checked()
+    print_separator("gcloud");
+
+    let output = ctx
+        .run_type()
+        .execute(&gcloud)
+        .args(["components", "update", "--quiet"])
+        .output()?;
+
+    if let ExecutorOutput::Wet(out) = output {
+        if !out.status.success() {
+            let msg = String::from_utf8(out.stderr)?;
+
+            if msg.contains("You cannot perform this action because the Google Cloud CLI component manager")
+                && msg.contains("is disabled for this installation")
+            {
+                println!("gcloud component manager is disabled, skipping.");
+                return Ok(());
+            }
+
+            return Err(eyre!(msg));
+        }
     }
```

Comment:
A few suggestions:
* Remove nesting with `Wet`
* Flush output (that way we do not need to print the output on error as well)
* Return `Err(SkipStep(...))` instead of `Ok`

See also how this was done for uv.
```suggestion
    let output = match output {
        ExecutorOutput::Wet(wet) => wet,
        ExecutorOutput::Dry => return Ok(()),
    }
    
    // When `gcloud` is installed via `apt`, the components update via `apt` as well
    let stderr = &output.stderr;
    if stderr.contains("You cannot perform this action because the Google Cloud CLI component manager")
                && stderr.contains("is disabled for this installation") {
        return Err(SkipStep("The Google Cloud CLI component manager is disabled for this installation"));
    }
    
    // Flush captured output
    std::io::stdout().write_all(&output.stdout)?;
    std::io::stderr().write_all(&output.stderr)?;
    
    if !output.status.success() {
        return Err(eyre!("gcloud component update failed"));
    }
```

Reply:
Also it looks like you might need `.to_string()`